server/src/main/java/org/elasticsearch/search/rank/RankSearchContext.java

```diff
@@ -0,0 +1,564 @@
+    @Override
+    public void groupStats(List<String> groupStats) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean version() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void version(boolean version) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean seqNoAndPrimaryTerm() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void seqNoAndPrimaryTerm(boolean seqNoAndPrimaryTerm) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public int[] docIdsToLoad() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public SearchContext docIdsToLoad(int[] docIdsToLoad) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public DfsSearchResult dfsResult() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void addDfsResult() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Use a separate query search result.
+     */
+    @Override
+    public QuerySearchResult queryResult() {
+        return querySearchResult;
+    }
+
+    @Override
+    public void addQueryResult() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public TotalHits getTotalHits() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public float getMaxScore() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public FetchPhase fetchPhase() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public FetchSearchResult fetchResult() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void addFetchResult() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Profiling is not supported by ranking.
+     */
+    @Override
+    public Profilers getProfilers() {
+        return null;
+    }
+
+    @Override
+    public long getRelativeTimeInMillis() {
+        return parent.getRelativeTimeInMillis();
+    }
+
+    @Override
+    public void registerAggsCollectorManager(CollectorManager<Collector, Void> collectorManager) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Aggregations are run as a separate query, so do not add any aggregations collectors.
+     */
+    @Override
+    public CollectorManager<Collector, Void> getAggsCollectorManager() {
+        return null;
+    }
+
+    @Override
+    public SearchExecutionContext getSearchExecutionContext() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public ReaderContext readerContext() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public SourceLoader newSourceLoader() {
+        throw new UnsupportedOperationException();
```

Comment:
so many methods! :) would it make sense to move up all the methods that do something and down all the methods that do nothing or throw? Maybe add a big comment that says "from now on, nothing to look at"

Reply:
Fixed.